### ISSUE TITLE
Add support for collecting usage data from plugin responses

### ISSUE THREAD

**Issue by willsmanley** [livekit/agents#613]
I need to track token usage on a per-user and per-session basis as I'm sure many others do as well.

It would be nice to configure llm like this:
```
def token_usage_callback(usage_data):
    print(usage_data)

llm = openai.llm(
   model="gpt-4o", 
   token_usage_callback=token_usage_callback
)
```

and then you can store all of that usage data as needed for analytics and attribution

**Comment by willsmanley** [comment]
hi @theomonnom what do you think about enabling usage attribution1 from plugins like this? is this already on your roadmap?

**Comment by davidzhao** [comment]
feature added in #984, will be released in 0.11.0

**Comment by davidzhao** [comment]
realtime plugin isn't exposing it yet.. tracking this effort in #1010
